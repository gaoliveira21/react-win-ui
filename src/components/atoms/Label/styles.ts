import styled, { css } from 'styled-components'

export const Label = styled.label`
  ${({ theme }) => css`
    font-size: ${theme.fonts.text.BODY.size};
    font-weight: ${theme.fonts.text.BODY.weight};
    line-height: ${theme.fonts.text.BODY.lineHeight};
    color: ${theme.colors.text.PRIMARY};
  `}
`
