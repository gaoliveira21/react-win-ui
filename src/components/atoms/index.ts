export * from './Text'
export * from './Heading'
export * from './Button'
export * from './Input'
export * from './Label'
