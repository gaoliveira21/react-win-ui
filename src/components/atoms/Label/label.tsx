import * as S from './styles'

export function Label(): React.ReactElement {
  return <S.Label>label</S.Label>
}
