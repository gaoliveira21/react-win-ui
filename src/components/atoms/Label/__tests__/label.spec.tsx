import { screen } from '@testing-library/react'

import { Label } from '@/components/atoms/Label'
import { renderWithTheme } from '@/tests/render-with-theme'

describe('<Label />', () => {
  it('Should render Label with correct text content', () => {
    const textContent = 'any_text'
    renderWithTheme(<Label>{textContent}</Label>)

    const label = screen.getByText(textContent)

    expect(label).toBeInTheDocument()
    expect(label).toMatchSnapshot()
  })
})
