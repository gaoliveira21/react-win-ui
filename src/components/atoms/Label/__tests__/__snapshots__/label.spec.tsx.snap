// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Should render Label with correct text content 1`] = `
.c0 {
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  color: rgb(0,0,0);
}

<label
  class="c0"
>
  any_text
</label>
`;
